core/block_validator.go

```diff
@@ -105,6 +105,24 @@ func (v *BlockValidator) ValidateBody(block *types.Block) error {
 
 		// The individual checks for blob validity (version-check + not empty)
 		// happens in StateTransition.
```

Comment:
According to the description of this function and this comment, the purpose of this function is mainly to validate the header of the given block. How about moving this validation to `func (st *StateTransition) preCheck()` this function?

Reply:
As I understand it, this part is not for validating the block's header. 

From the comment ,I understood this comment as verifying the transactions included in the block header.
Since the function name is `ValidateBody`, it is intended for validating the block's contents, and it seems fit to validate DepositTx here.

Plus, the `preCheck()` does not contain a for loop iterating transaction list, which is why i put code this function.

Reply:
```go
// BlockValidator is responsible for validating block headers, uncles and
// processed state.
```
This is the comment for `BlockValidator`, and I think `ValidateBody` does validate if the transactions root is correct with the txs in block header, and the gas usage is correct with the txs in block header.. which means it is for the integrity of header data with inner transactions.

Also `preCheck()` is for a single transaction, so you don't need to add a loop inside it.

Anyway it's just ok to locate this code block here logically, I thought it would be better to move it in the aspect of semantic. It was just a suggestion.

Reply:
But if we use preCheck(), we have to convert it to a message using `TransactionToMessage()`, but this function does not have any evm context, so it cannot determine whether the current block is after KromaMPT or not, and this is what I wanted to say.
I think I will have to find another good place.

Reply:
You mean `st.msg` does not have original transaction data, so we cannot distinguish if it is `DepositTx` or `KromaDepositTx` in `preCheck()`, right? I understand, then it is proper workaround to get it effectively.

Reply:
How about putting it to `state_processor.go:87`?

Reply:
How about checking out the beginning of the applyTransaction function in state_processor.go? 
we have both a message and an evm context, and a nice set of conditional statements based on the chain config.